pkg/datastore/repository.go

```diff
@@ -28,6 +28,7 @@ type AddressRepository interface {
 type CheckedHeadersRepository interface {
 	MarkHeaderChecked(headerID int64) error
 	MarkHeadersUnchecked(startingBlockNumber int64) error
```

Comment:
the new function has me thinking maybe we should name this `MarkHeadersUncheckedSince` to communicate how much it does - wouldn't want to call this one by accident!

Reply:
Yep, definitely. That was partially why I named the new on `MarkSingleHeaderUnchecked`.

Reply:
Yep definitely! That's partially why I named this new method `MarkSingleHeaderUnchecked`. But I'll update this method too, just to be explicit.